packages/engine-server/src/workspace/service.ts

```diff
@@ -864,12 +866,33 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
             return { repo, vaults, status: SyncActionStatus.NO_UPSTREAM };
           if (!(await this.shouldVaultsSync("pull", rootVaults)))
             return { repo, vaults, status: SyncActionStatus.SKIP_CONFIG };
-          if (await git.hasChanges({ untrackedFiles: "no" }))
-            return {
-              repo,
-              vaults,
-              status: SyncActionStatus.UNCOMMITTED_CHANGES,
-            };
+
+          let stashed: string | undefined;
+          if (await git.hasChanges({ untrackedFiles: "no" })) {
+            // stash away changes, otherwise pull will fail
+            try {
+              stashed = await git.stashCreate();
+              this.logger.info({ ctx, vaults, repo, stashed });
+              // this shouldn't fail, but for safety's sake
+              if (_.isEmpty(stashed) || !git.isValidStashCommit(stashed)) {
+                throw new DendronError({
+                  message: "unable to stash changes",
+                  payload: { stashed },
+                });
+              }
+              // stash create doesn't change the working directory, so we need to get rid of the tracked changes
+              await git.reset("hard");
```

Comment:
just double checking - but a user won't lose files that are covered under `.gitignore` in this case right? 

To me looks good, but just trying to think of any edge cases where someone may lose a file.

Reply:
`git reset --hard` doesn't touch untracked files (new files that haven't been `git add`ed) or ignored files, only tracked files.

The only risk of losing files I think would be if Dendron crashes after reset and before restore. If that does happen though, I immediately log the stash commit before resetting, so we could direct the user to read the log and `git stash apply ...` themselves.

I guess it would be a good idea to document that gotcha somewhere just in case someone hits it. I'll add a docs PR and link it in the main body of this PR.